### PR TITLE
Fix Small Menu Nav

### LINKED ISSUE #10: Remove Tests Link from the Menu when Screensize is Small
# Issue Description
The tests navigation link is suppose to be hidden 😄 

The navigation supplied to the application menu needs updated before being passes. Needs fixed [here](https://github.com/cerner/terra-site/blob/master/src/app/configureApp.jsx#L79)

### PR DESCRIPTION
**Pull request by emilyrohrbough**
### Summary
Closes #10 

## To Verify:
1. Go to https://github.com/cerner/terra-framework/pull/59
2. Shirk screens size to smaller viewports to access small navigation

### CODE REVIEW

File: src/app/ApplicationMenu.jsx [hunk 1]
```diff
@@ -22,19 +22,15 @@ const propTypes = {
 
 const defaultProps = {
   menuHeader: null,
-  navigation: undefined,
+  links: undefined,
```
Comment (JakeLaCombe) [inline src/app/ApplicationMenu.jsx]: For prop types of array, it always feels better to default it to an empty array as opposed to undefined. Do others have opinions on that?

File: src/app/common/menu/MenuList.jsx [hunk 2]
```diff
@@ -42,11 +41,11 @@ const propTypes = {
 const generateMenuLinks = (links, location) => {
   const navLinks = !!links && links.map(link =>
     <List.Item
-      key={link.id}
+      key={link.path}
```
Comment (JakeLaCombe) [inline src/app/common/menu/MenuList.jsx]: Should these keys be different? Feels like on key could be the id, and the other could be the path.

File: src/app/ApplicationMenu.jsx (hunk 1 above)
Comment (StephenEsser) [reply src/app/ApplicationMenu.jsx]: My preference is not to default the array. Defaulting creates an additional state the prop can be in during logic statements which can cause interesting situations.

| Syntax        | if (prop)           |
| ------------- |:-------------:|
| `<Component prop={['1', '2', '3']} />`     | true|
| `<Component />`      | true      |
| `<Component prop={null} />`  | false      |

File: src/app/common/menu/MenuList.jsx (hunk 2 above)
Comment (emilyrohrbough) [reply src/app/common/menu/MenuList.jsx]: The id is no longer being generated. Before it was generated [here](https://github.com/cerner/terra-site/pull/13/files#diff-2e239fdb31751e56fd0b23b7a78e5d13L33) which just set the path as the id.

**Pull request merged**